# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerChild.cpp
# Commit: 5d39af040a61
# Full Hash: 5d39af040a61b1baea92d2473ae22f9024f0dc1c
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-26 09:13:23
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r b9daa963dfd0 -r 5d39af040a61 dom/workers/remoteworkers/RemoteWorkerDebuggerChild.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerChild.cpp	Tue Mar 25 14:33:27 2025 +0000
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "RemoteWorkerDebuggerChild.h"
+#include "mozilla/dom/WorkerCommon.h"
+#include "mozilla/dom/WorkerPrivate.h"
+
+namespace mozilla::dom {
+
+RemoteWorkerDebuggerChild::RemoteWorkerDebuggerChild(
+    WorkerPrivate* aWorkerPrivate)
+    : mWorkerPrivate(aWorkerPrivate) {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(mWorkerPrivate);
+  mWorkerPrivate->AssertIsOnWorkerThread();
+}
+
+RemoteWorkerDebuggerChild::~RemoteWorkerDebuggerChild() {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(!mWorkerPrivate);
+}
+
+mozilla::ipc::IPCResult RemoteWorkerDebuggerChild::RecvInitialize(
+    const nsString& aURL) {
+  // Send a WorkerDebuggerRunnable to compile debugger script.
+  Unused << SendSetAsInitialized();
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult RemoteWorkerDebuggerChild::RecvPostMessage(
+    const nsString& aMessage) {
+  // Send a WorkerDebuggerRunnable to fire a message event on DebuggerGlobal.
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult RemoteWorkerDebuggerChild::RecvSetDebuggerReady(
+    const bool& aReady) {
+  // Should call mWorkerPrivate->SetIsDebuggerReady();
+  return IPC_OK();
+}
+
+}  // namespace mozilla::dom